docs/src/development/proposals/MEP9/README.md

```diff
@@ -0,0 +1,28 @@
+# No Open Ports To the Data Center
+
+Our metal-stack partitions typically have open ports for metal-stack native services, these are:
+
+- SSH port on the firewalls
+- bmc-reverse-proxy for serial console access through the metal-console
+
+Primarily the reason to get rid off these open ports is the reduction of the attack surface to the data center.
+
+As a next step, we can also cosider joining the management servers to the VPN mesh, which would replace typical Wireguard setups for operators to enter resources inside the partition.
+
+[](./architecture.drawio.svg)
+
+## Firewalls
+
+### Implementation
+
+- Add a headscale server to the metal control plane
+- The firewall allocation process in the metal-api will change the following way
+  1. Request a one-time joining token for the firewall VPN from the headscale gRPC endpoint (1 hour expiration time)
+  1. Inject the tailscaled service configuration into the userdata and launch it through ignition (along with user-given userdata)
```

Comment:
To me this approach feels wrong for following reasons:

- The userdata is owned by the user (as the name already tells), the provider should not inject something after that
- The provided userdata can be cloudinit or ignition format, it would require to distinguish between those formats for proper injection?
- The user is also free to just provide an arbitrary string without any format (it's the user's responsibility to provide a working configuration), how do you want to inject this to something like this?

Even if it allows no backwards compatibility, I find that a more adequate way to add this feature would include a change on the metal-images (like installing tailscaled on the image and enable through `install.sh` when a joining key is given).

Reply:
And joining key is part of the allocation. Agree, but we keep install.sh out of the game please, we could implement a service which picks the joining key from /etc/metal/install.yaml an generate a proper systemd service config for tailscaled and friends.

Reply:
Totally fine with this. Maybe we should raise an issue in metal-images with long-term objective to reduce logic in `install.sh`? Then we can discuss there how the current situation can be approved and start migrating things into such a service or whatever a better solution would look like.

Reply:
I created an issue here: https://github.com/metal-stack/metal-images/issues/140